scripts/linters/html_linter.py

```diff
@@ -110,6 +110,41 @@ def handle_starttag(self, tag, attrs):
             column_number + len(tag) + 2)
         starttag_text = self.get_starttag_text()
 
+        # Check whether there is space around attributes.
+        # An = is followed either by " or {.
+        tag_substrings = starttag_text.split()
```

Comment:
> How about the following ideas:

> 1. Splitting starttag_text line(s) with space and none of the elements should start or end with =?
> 2. For each [attr, value] pair, the starttag_text should contain attr=value string?

Any specific reason to follow the first idea and not the 2nd one?

Reply:
The first idea was chosen over the second was to mainly identify if there is space around, after, or before '='. I thought it gives a better picture of what the error is.

Reply:
The 2nd idea gives us a specific attribute where the extra space is provided, which can help us provide a better error message, right? But I'm not sure whether it will work for all the cases. Can you please compare both the ideas and write pros and cons? I can help you decide the better one! :)

Reply:
@DubeySandeep Actually, when I tried implementing it. It is more readable and maintainable than the 1st idea. But there is an edge case for 2nd idea for which I need your input. It is `[innerHTML]` in the code whereas `[innerhtml]`(lowercased attr names) when we get it as a parsed [attr, value] pair. Hence search wouldn't find anything. To avoid this issue, I thought of using a case insensitive search. Is that fine? or should I find another way?

Reply:
>  I thought of using a case insensitive search

It won't work correctly for `<tag-name attR = "value" attr="value"></tag-name>`, right? (though there is 0 chance of such code to exist but we should consider out functionality to work correctly on worst conditions.) 

Have you checked why/how the `attr` is converted to lower case?